### ISSUE TITLE
Crash during extended command line run

### ISSUE THREAD

**Issue by GWustenstein** [uhppoted/uhppoted-tunnel#4]
Endpoint running as follows:

sudo /home/pi/uhppoted/uhppoted-tunnel --console --max-retries -1 --max-retry-delay 5m --out udp/broadcast:255.255.255.255:60000 --in tls/client:[redacted] --ca-cert /home/pi/uhppoted/actrl-ca-crt.pem --cert /home/pi/uhppoted/actrl-ctrl-crt.pem --key /home/pi/uhppoted/actrl-ctrl-key.pem

humming along like this last line for random number of hours then.., 
(crash leaves the .pid file behind, btw)

2022/10/02 17:18:12 INFO   TLS    msg 115468  sent 360 bytes to [redacted]
fatal error: concurrent map read and map write
fatal error: concurrent map read and map write

goroutine 31512 [running]:
runtime.throw({0x3c4576, 0x21})
	runtime/panic.go:992 +0x5c fp=0x1d56ee0 sp=0x1d56ecc pc=0x47bf8
runtime.mapaccess2_fast32(0x378680, 0x1c00480, 0x1c312)
	runtime/map_fast32.go:62 +0x194 fp=0x1d56ef0 sp=0x1d56ee0 pc=0x202ac
github.com/uhppoted/uhppoted-tunnel/router.(*Router).get(0x625478, 0x1c312)
	github.com/uhppoted/uhppoted-tunnel/router/router.go:92 +0x3c fp=0x1d56f50 sp=0x1d56ef0 pc=0x2b74c8
github.com/uhppoted/uhppoted-tunnel/router.(*Switch).Received(0x1c0e968, 0x1c312, {0x1cf4006, 0x1c, 0x7fa}, 0x1ca8648)
	github.com/uhppoted/uhppoted-tunnel/router/router.go:62 +0x24 fp=0x1d56f64 sp=0x1d56f50 pc=0x2b7108
github.com/uhppoted/uhppoted-tunnel/tunnel/tls.(*tlsClient).received(0x1cac050, {0x1cf4000, 0x22, 0x800}, 0x1c0e968, {0x4415b4, 0x1cc6000})
	github.com/uhppoted/uhppoted-tunnel/tunnel/tls/tls_client.go:162 +0x128 fp=0x1d56fcc sp=0x1d56f64 pc=0x339c24
github.com/uhppoted/uhppoted-tunnel/tunnel/tls.(*tlsClient).listen.func1()
	github.com/uhppoted/uhppoted-tunnel/tunnel/tls/tls_client.go:150 +0x54 fp=0x1d56fec sp=0x1d56fcc pc=0x339a88
runtime.goexit()
	runtime/asm_arm.s:824 +0x4 fp=0x1d56fec sp=0x1d56fec pc=0x7a140
created by github.com/uhppoted/uhppoted-tunnel/tunnel/tls.(*tlsClient).listen
	github.com/uhppoted/uhppoted-tunnel/tunnel/tls/tls_client.go:149 +0xfc

goroutine 1 [chan receive, 509 minutes]:
github.com/uhppoted/uhppoted-tunnel/commands.(*Run).run(0x625900, 0x1ca80d8, {0x440ad8, 0x1c50180}, 0x1c0e6a8, 0x1c9a200)
	github.com/uhppoted/uhppoted-tunnel/commands/run.go:259 +0x228
github.com/uhppoted/uhppoted-tunnel/commands.(*Run).exec(0x625900, 0x1ca80d8, {0x440ad8, 0x1c50180}, 0x1c0e6a8)
	github.com/uhppoted/uhppoted-tunnel/commands/run_linux.go:66 +0x27c
github.com/uhppoted/uhppoted-tunnel/commands.(*Run).Execute.func1(0x1ca80d8, {0x440ad8, 0x1c50180}, 0x1c0e6a8)
	github.com/uhppoted/uhppoted-tunnel/commands/run_linux.go:33 +0x3c
github.com/uhppoted/uhppoted-tunnel/commands.(*Run).execute(0x625900, 0x1d11f50)
	github.com/uhppoted/uhppoted-tunnel/commands/run.go:189 +0xa88
github.com/uhppoted/uhppoted-tunnel/commands.(*Run).Execute(0x625900, {0x0, 0x0, 0x0})
	github.com/uhppoted/uhppoted-tunnel/commands/run_linux.go:36 +0x100
main.main()
	github.com/uhppoted/uhppoted-tunnel/cmd/uhppoted-tunnel/main.go:31 +0x114

goroutine 4 [select]:
github.com/uhppoted/uhppoted-tunnel/router.init.0.func1()
	github.com/uhppoted/uhppoted-tunnel/router/router.go:44 +0x9c
created by github.com/uhppoted/uhppoted-tunnel/router.init.0
	github.com/uhppoted/uhppoted-tunnel/router/router.go:38 +0x1c

goroutine 20 [select, 509 minutes]:
github.com/uhppoted/uhppoted-tunnel/tunnel/tls.(*tlsClient).connect.func1()
	github.com/uhppoted/uhppoted-tunnel/tunnel/tls/tls_client.go:109 +0x154
created by github.com/uhppoted/uhppoted-tunnel/tunnel/tls.(*tlsClient).connect
	github.com/uhppoted/uhppoted-tunnel/tunnel/tls/tls_client.go:107 +0x218

goroutine 8 [syscall, 509 minutes]:
os/signal.signal_recv()
	runtime/sigqueue.go:151 +0x34
os/signal.loop()
	os/signal/signal_unix.go:23 +0x14
created by os/signal.Notify.func1.1
	os/signal/signal.go:151 +0x28

goroutine 9 [select, 509 minutes]:
github.com/uhppoted/uhppoted-tunnel/tunnel.(*Tunnel).Run(0x1ca80d8, 0x1c9a200)
	github.com/uhppoted/uhppoted-tunnel/tunnel/tunnel.go:66 +0x2e8
github.com/uhppoted/uhppoted-tunnel/commands.(*Run).run.func1()
	github.com/uhppoted/uhppoted-tunnel/commands/run.go:254 +0x64
created by github.com/uhppoted/uhppoted-tunnel/commands.(*Run).run
	github.com/uhppoted/uhppoted-tunnel/commands/run.go:252 +0x214

...and the dump goes on, and on, and on...

**Comment by twystd** [comment]
Missing _mutex_ in `Router::get`:
```
func (r *Router) get(id uint32) func([]byte) {
	if h, ok := r.handlers[id]; ok && h.f != nil {
		h.touched = time.Now()
		return h.f
	}

	return nil
}
```

**Comment by twystd** [comment]
Fix in place - running extended duration test but should be good to go.

**Comment by uhppoted** [comment]
Ran for 8 hours yesterday under moderate load without glitching - am going to call this fixed.

Leaving this open pending release.

**Comment by uhppoted** [comment]
Closing - fix released in [v0.8.2](https://github.com/uhppoted/uhppoted-tunnel/releases/tag/v0.8.2). 

Feel free to reopen if necessary :-).
